packages/from-hardhat/src/config.ts

```diff
@@ -0,0 +1,36 @@
+import type { HardhatConfig } from "hardhat/types";
+
+import Config from "@truffle/config";
+
+export const fromHardhatConfig = (hardhatConfig: HardhatConfig): Config => {
+  return Config.default().merge({
+    networks: Networks.fromHardhatConfig(hardhatConfig)
+  });
+};
+
+namespace Networks {
+  export const fromHardhatConfig = (hardhatConfig: HardhatConfig) =>
+    Object.entries(hardhatConfig.networks)
+      .flatMap(([networkName, networkConfig]) => {
+        // exclude hardhat network as not supported
+        if (networkName === "hardhat") {
+          return [];
+        }
+
+        // only accept netowrks that specify `url`
+        if (!networkConfig || !("url" in networkConfig)) {
+          return [];
+        }
+
+        const { url } = networkConfig;
+        return [
+          {
+            [networkName]: {
+              url,
+              network_id: "*"
+            }
+          }
+        ];
+      })
+      .reduce((a, b) => ({ ...a, ...b }), {});
```

Comment:
I wonder if there's some more readable way to write this line specifically? I was initially thinking maybe `Object.assign` with a spread operator, but I dunno, the function chaining does kinda win out here over nested calls, IMO.

Reply:
`Object.assign()` is just ugly... this is my favorite way to do this, but nowadays maybe there's other sugar to consider? Happy to take suggestions.

Reply:
Nah don't worry about it. I was just wondering aloud here.